test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go

```diff
@@ -50,38 +52,100 @@ var _ = SIGDescribe("[HPA] Horizontal pod autoscaling (scale resource: Custom Me
+			pod:        monitoring.StackdriverExporterPod(stackdriverExporterPod, f.Namespace.Name, stackdriverExporterPod, monitoring.CustomMetricName, metricValue),
+			hpa:        objectHPA(f.Namespace.ObjectMeta.Name, metricTarget)}
+		tc.Run()
+	})
+
+	It("should scale down with External Metric with target value from Stackdriver [Feature:CustomMetricsAutoscaling]", func() {
+		initialReplicas := 2
+		// metric should cause scale down
+		metricValue := externalMetricValue
+		metricTarget := 2 * int64(initialReplicas) * metricValue
+		metricTargets := map[string]externalMetricTarget{
+			"target": {
+				value:     metricTarget,
+				isAverage: false,
+			},
+		}
+		tc := CustomMetricTestCase{
+			framework:       f,
+			kubeClient:      f.ClientSet,
+			initialReplicas: initialReplicas,
+			scaledReplicas:  1,
+			//
+			deployment: monitoring.SimpleStackdriverExporterDeployment(dummyDeploymentName, f.Namespace.ObjectMeta.Name, int32(initialReplicas), 0 /* ignored */),
+			pod:        monitoring.StackdriverExporterPod(stackdriverExporterPod, f.Namespace.Name, stackdriverExporterPod, "target", metricValue),
+			hpa:        externalHPA(f.Namespace.ObjectMeta.Name, metricTargets)}
+		tc.Run()
+	})
+
+	It("should scale down with External Metric with target average value from Stackdriver [Feature:CustomMetricsAutoscaling]", func() {
+		initialReplicas := 2
+		// metric should cause scale down
+		metricValue := externalMetricValue
+		metricAverageTarget := 2 * metricValue
+		metricTargets := map[string]externalMetricTarget{
+			"target_average": {
+				value:     metricAverageTarget,
+				isAverage: true,
+			},
+		}
+		tc := CustomMetricTestCase{
+			framework:       f,
+			kubeClient:      f.ClientSet,
+			initialReplicas: initialReplicas,
+			scaledReplicas:  1,
+			//
+			deployment: monitoring.SimpleStackdriverExporterDeployment(dummyDeploymentName, f.Namespace.ObjectMeta.Name, int32(initialReplicas), 0 /* ignored */),
+			pod:        monitoring.StackdriverExporterPod(stackdriverExporterPod, f.Namespace.Name, stackdriverExporterPod, "target_average", externalMetricValue),
+			hpa:        externalHPA(f.Namespace.ObjectMeta.Name, metricTargets)}
+		tc.Run()
 	})
 
 	It("should scale down with Custom Metric of type Pod from Stackdriver with Prometheus [Feature:CustomMetricsAutoscaling]", func() {
 		initialReplicas := 2
-		scaledReplicas := 1
 		// metric should cause scale down
 		metricValue := int64(100)
 		metricTarget := 2 * metricValue
-		deployment := monitoring.PrometheusExporterDeployment(stackdriverExporterDeployment, f.Namespace.ObjectMeta.Name, int32(initialReplicas), metricValue)
-		customMetricTest(f, f.ClientSet, simplePodsHPA(f.Namespace.ObjectMeta.Name, metricTarget), deployment, nil, initialReplicas, scaledReplicas)
+		tc := CustomMetricTestCase{
+			framework:       f,
+			kubeClient:      f.ClientSet,
+			initialReplicas: initialReplicas,
+			scaledReplicas:  1,
+			// Metric exported by deployment is ignored
```

Comment:
Where's the metric value coming from? There's no exporter pod except for deployment in this test case.

Reply:
There is an exporter pod in this test case

Reply:
The comment was a lie, we use the metric from deployment. Deleted.

Reply:
Right, I think that comment was supposed to be added in another place. Sorry!